Sources/Workspace/Workspace.swift

```diff
@@ -2561,168 +2684,208 @@ extension Workspace {
 
-                // If the dependency is present, we're done.
-                let dependencyPath = self.path(to: dependency)
-                guard !fileSystem.isDirectory(dependencyPath) else { return }
+    /// Create a loadable result.
+    public init(_ construct: @escaping () throws -> Value) {
+        self.construct = construct
+    }
 
-                switch dependency.state {
-                case .checkout(let checkoutState):
-                    // If some checkout dependency has been removed, clone it again.
-                    _ = try clone(package: dependency.packageRef, at: checkoutState)
-                    diagnostics.emit(.checkedOutDependencyMissing(packageName: dependency.packageRef.name))
+    /// Load and return the result.
+    public func loadResult() -> Result<Value, Error> {
+        return Result(catching: {
+            try self.construct()
+        })
+    }
 
-                case .edited:
-                    // If some edited dependency has been removed, mark it as unedited.
-                    //
-                    // Note: We don't resolve the dependencies when unediting
-                    // here because we expect this method to be called as part
-                    // of some other resolve operation (i.e. resolve, update, etc).
-                    try unedit(dependency: dependency, forceRemove: true, diagnostics: diagnostics)
+    /// Load and return the value.
+    public func load() throws -> Value {
+        return try loadResult().get()
+    }
+}
 
-                    diagnostics.emit(.editedDependencyMissing(packageName: dependency.packageRef.name))
+private struct RemoteArtifact {
+    let packageRef: PackageReference
+    let targetName: String
+    let url: Foundation.URL
+    let checksum: String
+}
 
-                case .local:
-                    self.state.dependencies.remove(dependency.packageRef.identity)
-                    try self.state.save()
-                }
-            }
+private struct ArchiveIndexFile: Decodable {
+    let schemaVersion: String
+    let archives: [Archive]
+
+    struct Archive: Decodable {
+        let fileName: String
+        let checksum: String
+        let supportedTriples: [Triple]
+
+        enum CodingKeys: String, CodingKey {
+            case fileName
+            case checksum
+            case supportedTriples
+        }
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            self.fileName = try container.decode(String.self, forKey: .fileName)
+            self.checksum = try container.decode(String.self, forKey: .checksum)
+            self.supportedTriples = try container.decode([String].self, forKey: .supportedTriples).map(Triple.init)
         }
     }
 }
 
-// MARK: - Repository Management
+// MARK: - Package container provider
 
-extension Workspace {
+extension Workspace: PackageContainerProvider {
+    public func getContainer(
```

Comment:
👀 this is the key change, it replaces RepositoryPackageContainerProvider which was removed